vignettes_src/troubleshooting.Rmd

```diff
@@ -0,0 +1,285 @@
+```
+
+```{r, include = FALSE, error = TRUE}
+task_wait(id)
+```
+
+Here is the error, which is a bit less informative this time:
+
+```{r}
+task_result(id)
+```
+
+The log gives a better idea of what is going on - the file
+`c:/myfile.csv` does not exist (because it is not found on the
+cluster; using relative paths is much preferred to absolute paths)
+
+```{r}
+task_log_show(id)
+```
+
+The real content of the error message is present in the warning!
+You can also get the warnings with
+
+```{r}
+task_result(id)$warnings
+```
+
+Which will be a list of all warnings generated during the execution
+of your task (even if it succeeds).  The traceback also shows what
+happened:
+
+```{r}
+task_result(id)$trace
+```
+
+## Caused by an error during startup
+
+These are harder to troubleshoot but we can still pull some information out.  The example here was a real-world case and illustrates one of the issues with using a shared filesystem in the way that we do here.
+
+```{r echo = FALSE, results = "hide"}
+writeLines(
+  c("times2 <- function(x) {",
+    "  2 * x",
+    "}"),
+  "mycode.R")
+```
+
+Suppose you have a context that uses some code in `mycode.R`:
+
+```{r echo = FALSE, results = "asis"}
+r_output(readLines("mycode.R"))
+```
+
+We can create an environment with this code and use it just fine:
+
+```{r}
+hipercow_environment_create(sources = "mycode.R")
+id <- task_create_expr(times2(10))
+task_wait(id)
+task_result(id)
+```
+
+...but imagine then you're editing the file and save the file but it is not
+syntactically correct:
+
+```{r echo = FALSE, results = "hide"}
+writeLines(
+  c("times2 <- function(x) {",
+    "  2 * x", "}",
+    "newfun <- function(x)"),
+  "mycode.R")
+```
+```{r echo = FALSE, results = "asis"}
+r_output(readLines("mycode.R"))
+```
+
+And then you either submit a task, **or** a task that you have
+previously submitted gets run (which could happen ages after you
+submit it if the cluster is busy).
+
+```{r}
+hipercow_environment_create(sources = "mycode.R")
+id <- task_create_expr(times2(10))
+task_wait(id)
+task_status(id)
+```
+
+The error here has happened before getting to your code - it is
+happening when context loads the source files.  The log makes this
+a bit clearer:
+
+```{r}
+task_log_show(id)
+```
+
+## My task got stuck at `submitted`
+
+(Previous users of `didehpc` may recognise this as being stuck at `PENDING`).
+
+This is the most annoying one, and can happen for many reasons. You can see via the [web interface](mrcdata.dide.ic.ac.uk/hpc/) or the Microsoft cluster tools that your task has failed but `hipercow` is reporting it as pending.  This happens when something has failed during the script that runs *before* any `hipercow` code runs on the cluster.
+
+Things that have triggered this situation in the past:
+
+* An error in the Microsoft cluster tools
+* A misconfigured node (sometimes they are missing particular software)
+* A networking issue
+* Gremlins
+* Network path mapping error
```

Comment:
* Running out of disk space at some point after submitting the job

Reply:
added